project-koku/nise
# Cost 2380 - OCI upload report to storage bucket

**Changes proposed**:

- add upload report to OCI storage bucket
- add oci python sdk
- update OCI report options in README and report usage docs
- update nise version

<br/>

**Testing Instructions**:

 - test command-line input

    Run this command:

        python3 -m unittest tests.test_main.CommandLineTestCase.test_valid_oci_input
        python3 -m unittest tests.test_main.CommandLineTestCase.test_valid_oci_inputs_no_config_file

 - test writing a report and routing it to the appropriate storage

    Run this command:

        python3 -m unittest tests.test_report.OCIReportTestCase


 - test uploading reports to OCI storage

    Run these command:

        python3 -m unittest tests.test_upload.UploadTestCase.test_upload_to_oci_bucket_success
        python3 -m unittest tests.test_upload.UploadTestCase.test_upload_to_oci_bucket_config_not_found
        python3 -m unittest tests.test_upload.UploadTestCase.test_upload_to_oci_bucket_invalid_config


For each of the above tests,  expect to see the number of tests run and the time taken in seconds;
    e.g.  `Ran 1 test in 0.006s`
